src/command.c

```diff
@@ -679,10 +687,11 @@ static bool cmd_heapinfo(target_s *t, int argc, const char **argv)
 		target_addr_t heap_limit = strtoul(argv[2], NULL, 16);
 		target_addr_t stack_base = strtoul(argv[3], NULL, 16);
 		target_addr_t stack_limit = strtoul(argv[4], NULL, 16);
-		gdb_outf("heapinfo heap_base: %p heap_limit: %p stack_base: %p stack_limit: %p\n", heap_base, heap_limit,
-			stack_base, stack_limit);
+		gdb_outf("heap_base: %08" PRIx32 " heap_limit: %08" PRIx32 " stack_base: %08" PRIx32 " stack_limit: %08" PRIx32
+				 "\n",
+			heap_base, heap_limit, stack_base, stack_limit);
 		target_set_heapinfo(t, heap_base, heap_limit, stack_base, stack_limit);
 	} else
-		gdb_outf("heapinfo heap_base heap_limit stack_base stack_limit\n");
+		gdb_outf("?\n");
```

Comment:
Rather than doing this which is.. cryptic.. at best and a bad change, please make this help string match the command usage string above - string folding will mean this uses no more Flash than the original string while also providing useful help to the user.